library/tuple.lisp

```diff
@@ -69,6 +69,12 @@
           (hash a)
           (hash b))))))
 
+  (define-instance (Bifunctor Tuple)
+    (define (bimap f g item)
+      (match item
+        ((Tuple a b)
+         (Tuple (f a) (g b))))))
```

Comment:
Haskell also provides instances for all the bigger `Tuple*` types. Should I do that here as well?

Reply:
Might as well just for completeness.

Reply:
I don't love that, because e.g. for `Tuple3`, the instance will be `Bifunctor (Tuple3 :a)`, right? And `first` will operate on the second element of the tuple, and `second` on the third? Feels a little bit wrong.

Reply:
I'd tend to agree with @gefjon . And at that point, shouldn't the user be making a custom type anyway? Why are they fiddling around with length-N tuples? I've never personally used larger than a size 3 or 4, and even then I never needed to map across it.